s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -478,13 +478,37 @@ func (t *TidbClient) CountObjects(ctx context.Context, bucketName, prefix string
 	return &rsp, err
 }
 
-func (t *TidbClient) DeleteBucket(ctx context.Context, bucket *Bucket) error {
+func (t *TidbClient) DeleteBucket(ctx context.Context, bucket *Bucket) (err error) {
+	var sqlTx *sql.Tx
+	var tx interface{}
+	tx, err = t.Client.Begin()
+
+	defer func() {
+		if err == nil {
+			err = sqlTx.Commit()
+		}
+		if err != nil {
+			sqlTx.Rollback()
+		}
+	}()
+
+	sqlTx, _ = tx.(*sql.Tx)
+
 	sqltext := "delete from buckets where bucketname=?;"
-	_, err := t.Client.Exec(sqltext, bucket.Name)
+	_, err = sqlTx.Exec(sqltext, bucket.Name)
 	if err != nil {
-		return handleDBError(err)
+		err = handleDBError(err)
```

Comment:
Don't we need to return here?

Reply:
to err on the side of caution (based on sample code i saw on a blog), did not return...yes, that might be more efficient in the sense we will not try the second SQL, but even with current code, the intent is surely met, either BOTH SQL are executed or NONE are executed. Tested this. Thanks